src/G2/Equiv/G2Calls.hs

```diff
@@ -330,6 +333,23 @@ recursionInCase (State { curr_expr = CurrExpr _ e }) =
             p == T.pack "REC" -- && containsCase sk
         _ -> False
 
+-- used by EquivADT and Tactics
+concretizable :: Type -> Bool
+concretizable (TyVar _) = False
+concretizable (TyForAll _ _) = False
+concretizable (TyFun _ _) = False
+concretizable t@(TyApp _ _) =
+  typeFullApp t && (concretizable $ last $ TY.unTyApp t)
+concretizable TYPE = False
+concretizable TyUnknown = False
+concretizable _ = True
+
+typeFullApp :: Type -> Bool
+typeFullApp t@(TyApp _ _) =
+    let c_unapp = length $ TY.unTyApp t
+    in c_unapp >= 2 && 1 + argCount t == c_unapp
+typeFullApp _ = False
```

Comment:
I don't understand this- what are the checks on length about?

Reply:
The value `argCount t` is the max number of arguments that the function type at the head of t's TyApp can take.  The value `c_unapp` is the number of arguments that are actually taken, plus one for the function type at the head.  We need to check that those are equal to confirm that t is a concretizable type since function types aren't concretizable.  The >= 2 part may be unnecessary, but I had a check like that in exprFullApp, and I included it here to be consistent.

Reply:
Should I just write a new helper function that does what I thought spArgumentTypes did, or do you think that this method of checking for "full applications" is bad overall?